sdks/android/Mentat/library/src/main/java/org/mozilla/mentat/RustObject.java

```diff
@@ -10,21 +10,32 @@
 
 package org.mozilla.mentat;
 
+import android.util.Log;
+
 import com.sun.jna.Memory;
 import com.sun.jna.Pointer;
 
-import java.io.Closeable;
 import java.nio.ByteBuffer;
 import java.util.UUID;
 
 /**
  * Base class that wraps an non-optional {@link Pointer} representing a pointer to a Rust object.
- * This class implements {@link Closeable} but does not provide an implementation, forcing all
+ * This class implements {@link AutoCloseable} but does not provide an implementation, forcing all
  * subclasses to implement it. This ensures that all classes that inherit from RustObject
  * will have their {@link Pointer} destroyed when the Java wrapper is destroyed.
  */
-abstract class RustObject implements Closeable {
-    Pointer rawPointer;
+abstract class RustObject implements AutoCloseable {
+    // This should probably be private to let us better prevent usage mistakes (which lead to
+    // memory-unsafety).
+    protected Pointer rawPointer;
+    // If this is set to true, we don't destroy this object. Right now this is used for TypedValueAtIndex
```

Comment:
I don't see another usage of `notOwned` in _this_ patch, but perhaps it is used in others.  Can we improve this, in any case, perhaps by changing `TypedValueAtIndex`?